src/main/java/org/apache/sling/jcr/repoinit/impl/AclUtil.java

```diff
@@ -152,7 +154,9 @@ private static void setAcl(Session session, List<String> principals, String jcrP
 
         boolean changed = false;
         for (String name : principals) {
-            Principal principal = AccessControlUtils.getPrincipal(session, name);
+            Principal principal = options.contains("ignoreMissingPrincipal")
+                    ? () -> name
```

Comment:
i would slightly alter the logic here, as the fallback below is significant..... if you just use the specified name now, you might break that contract below.

so, i would suggest:
- always try to retrieve the principal first
- if it does not exist -> try backwards compatible fallback to original code
- if it does not exist -> create the anonymous principal in case option is set as you do here, otherwise use old behavior.

i believe this way we run the least risk of introducing subtle regressions that are super hard to troubleshoot.

Reply:
Sounds sensible. Thanks for catching this change in logic!